game/scripts/vscripts/components/duels/duels.lua

```diff
@@ -95,10 +95,11 @@ function Duels:Init ()
         -- player is not in a duel, they can just chill tf out
         return
       end
-      player.disconnected = false
       if player.killed or not player.disconnected then
         return
       end
+      player.disconnected = false
```

Comment:
What is this `player.disconnected` even for? I doubt it's persistent since I think player instances are pretty much just deleted when players disconnect.

Reply:
The player variable here is a reference to the player object within the currentDuel. It uses the disconnected vs killed flag to tell the difference between a player that died before/after disconnect to keep track of the score correctly. There was an error in the code that always counted reconnecting players as having not had disconnected so it wasn't un-counting and unfreezing them (also I had forgotten to add the unfreeze code).

Reply:
Oh wow. It's not a player instance. I saw `Duels:AllPlayers` returning a player instance in one branch and didn't look further than that. This code confuses me so hard.

Reply:
It's bad code. One of the first features written, there's a ticket to refactor it open for like 8+ months xD